src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrResult = [];
+  const STATE_PREVIOS = { ...state };
+
+  for (const OBJ_ACTION of actions) {
+    if (OBJ_ACTION.type === 'addProperties') {
+      Object.assign(STATE_PREVIOS, OBJ_ACTION.extraData);
+      arrResult.push({ ...STATE_PREVIOS });
+    }
+
+    if (OBJ_ACTION.type === 'removeProperties') {
+      for (const key of OBJ_ACTION.keysToRemove) {
+        delete STATE_PREVIOS[key];
+      }
+      arrResult.push({ ...STATE_PREVIOS });
+    }
+
+    if (OBJ_ACTION.type === 'clear') {
+      for (const key in STATE_PREVIOS) {
+        delete STATE_PREVIOS[key];
+      }
+      arrResult.push({ ...STATE_PREVIOS });
```

Comment:
Instead of using multiple `if` statements, consider using a `switch` statement to handle different action types, as it is more suitable for this kind of conditional logic where you have a limited number of conditions to check.